### ISSUE TITLE
Attack self account with gold pillaged 

### ISSUE THREAD

**Issue by LordMike13** [OpenThrone/OpenThrone#261]
attack with my own account and defense with my own account, loss of gold
![image](https://github.com/user-attachments/assets/0284fc1b-a2c5-4ed9-a37e-7d6a9c912d2f)
![image](https://github.com/user-attachments/assets/b9e09d35-9045-4ff5-9ce1-3c3ca0e7ae72)

**Comment by LordMike13** [comment]
![image](https://github.com/user-attachments/assets/523331da-1722-47d2-a872-9f5758c22f70)
![image](https://github.com/user-attachments/assets/6b51d033-0a30-4691-a26a-28e193fa291e)

**Comment by uaktags** [comment]
Do you have any information on how you initiated the attacks on self?

**Comment by uaktags** [comment]
I'm not worried about the fact that you lose gold on the attack, and its easy enough to add a block into the api. I just need to know if this is a front end bug. If you just hit the api, that's whatever, but i need more information to really be able to work on this.

**Comment by LordMike13** [comment]
Hello, thank you for your prompt response,
The self-attack happened at intervals while attacking other users.

**Comment by uaktags** [comment]
I'm not sure i fully follow/understand. The attack logs, while they show within the same minute, they don't appear within the same time. The logic also requires a POST call to the api which is not part of any of the internal code outside of the attack modal from the user profile.

The first instance of you attacking yourself was 9731 which happened 5seconds after 9730. did you click anything after the battle results page of 9730 loaded? Are you loading up any other pages? Tabs? Etc

**Comment by LordMike13** [comment]
Hello, first normal attack and when I attacked it for the second time the attack button appeared again and in the second attack is where that autoattack happened
attack again button, then pum...autoattack

**Comment by Revenate39** [comment]
I ******** me up!!!
I think it is the 'Attack Back' Button.
So select whoever to attack, attack them is fine, it's when wanting to attack again, and clicking the Attack Back, somehow the game allows the user to self attack
![image](https://github.com/user-attachments/assets/2a7eedaa-611e-42de-bdd5-2a9ae383ee1e)

**Comment by uaktags** [comment]
Thanks, i fixed the ability to attack yourself, but will now look into the button! Appreciate it. I'll get this pushed out shortly. Surprised this is the first we've seen of it.

**Comment by Revenate39** [comment]
It used to say 'Attack Again' I'm fairly sure, and definitely worked correctly in the past.

**Comment by uaktags** [comment]
Yep, was an error with a change from a static page to a server side generated one. I originally built statically compiled results so that the first time they're displayed, they'd be cached and would ease the load on the DB by serving the data from the server's filesystem. However, the impact to the DB was minimal and was interfering with a number of other checks we needed to get ACL/Alliance Sharing to work. So I ported the design to server-side generation and forgot to add the "viewerID" which lets us know if the person viewing the page is the attacker or the defender.

whoopsie
